### PR TITLE
Update plugins to use IPC RPC server - Closes #7133

### LINKED ISSUE #7133: Update plugin to use IPC endpoint instead of channel between application
### Description
- Update application to throw error if IPC is not enabled and plugin is registered
- Update plugin to use API client to connect to the RPC endpoint given by the config
- Remove events from the plugin

### Acceptance Criteria
- All tests for the plugin should be passing

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #7133

### How was it solved?

- Fix RPC server implementation
- Fix RPC server events
- Update RPC definitions
- Update APIClient to the latest structure
    - Update to use metadata/schema
    - Update the expected reply from RPC server
- Update transaction library to accept undefined schema

### How was it tested?

- Update all the tests
- Use example dpos app to run with plugins, such as report misbehavior command

### CODE REVIEW

File: framework/src/engine/endpoint/txpool.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	private readonly _pool: TransactionPool;
+	private readonly _broadcaster: Broadcaster;
+
+	public constructor(args: EndpointArgs) {
+		this._abi = args.abi;
+		this._pool = args.pool;
+		this._broadcaster = args.broadcaster;
+	}
+
+	public async postTransaction(ctx: RequestContext): Promise<PostTransactionResponse> {
```
Comment (shuse2) [inline framework/src/engine/endpoint/txpool.ts:47]: moved `postTransaction` and `getTransactionsFromPool` to `txpool` namespace

File: framework/src/engine/engine.ts [hunk 2]
```diff
@@ -220,21 +233,31 @@ export class Engine {
...
 		for (const [name, handler] of Object.entries(getEndpointHandlers(systemEndpoint))) {
 			this._rpcServer.registerEndpoint('system', name, handler);
 		}
+		for (const [name, handler] of Object.entries(getEndpointHandlers(txpoolEndpoint))) {
+			this._rpcServer.registerEndpoint('txpool', name, handler);
+		}
 		for (const [name, handler] of Object.entries(getEndpointHandlers(this._generator.endpoint))) {
 			this._rpcServer.registerEndpoint('generator', name, handler);
 		}
+		for (const [name, handler] of Object.entries(getEndpointHandlers(this._network.endpoint))) {
```
Comment (shuse2) [inline framework/src/engine/engine.ts:254]: recovered missing endpoint

File: framework/src/engine/engine.ts [hunk 3]
```diff
@@ -243,6 +266,50 @@ export class Engine {
 		);
 	}
 
+	private _registerEventListeners() {
```
Comment (shuse2) [inline framework/src/engine/engine.ts:269]: Recovered missing events.

**Review by Incede — APPROVED** [review]

File: elements/lisk-api-client/src/types.ts [hunk 4]
```diff
@@ -175,15 +181,103 @@ export interface PeerInfo {
 	readonly options?: { [key: string]: unknown };
 }
 
-export interface Block<T = Buffer | string> {
-	header: {
-		[key: string]: unknown;
-		id?: T;
-		version: number;
+type Primitive = string | number | bigint | boolean | null | undefined;
+type Replaced<T, TReplace, TWith, TKeep = Primitive> = T extends TReplace | TKeep
```
Comment (ishantiw) [inline elements/lisk-api-client/src/types.ts:185]: 🎉

File: framework-plugins/lisk-framework-monitor-plugin/src/controllers/prometheus.ts [hunk 5]
```diff
@@ -61,9 +61,9 @@ export const getData = (client: BasePlugin['apiClient'], state: SharedState) =>
 	next: NextFunction,
 ): Promise<void> => {
 	try {
-		const connectedPeers: PeerInfo[] = await client.invoke('app_getConnectedPeers');
-		const disconnectedPeers: PeerInfo[] = await client.invoke('app_getDisconnectedPeers');
-		const nodeInfo: NodeInfo = await client.invoke('app_getNodeInfo');
+		const connectedPeers: PeerInfo[] = await client.invoke('network_getConnectedPeers');
+		const disconnectedPeers: PeerInfo[] = await client.invoke('network_getDisconnectedPeers');
+		const nodeInfo: NodeInfo = await client.invoke('network_getNodeInfo');
```
Comment (ishantiw) [inline framework-plugins/lisk-framework-monitor-plugin/src/controllers/prometheus.ts:66]:
```suggestion
		const nodeInfo: NodeInfo = await client.invoke('system_getNodeInfo');
```

File: framework-plugins/lisk-framework-monitor-plugin/test/unit/prometheus.spec.ts [hunk 6]
```diff
@@ -126,11 +126,11 @@ describe('networkStats', () => {
...
 
 		when(channelInvokeMock)
-			.calledWith('app_getConnectedPeers')
+			.calledWith('network_getConnectedPeers')
 			.mockResolvedValue(connectedPeers)
-			.calledWith('app_getDisconnectedPeers')
+			.calledWith('network_getDisconnectedPeers')
 			.mockResolvedValue(disconnectedPeers)
-			.calledWith('app_getNodeInfo')
+			.calledWith('network_getNodeInfo')
```
Comment (ishantiw) [inline framework-plugins/lisk-framework-monitor-plugin/test/unit/prometheus.spec.ts:133]:
```suggestion
			.calledWith('system_getNodeInfo')
```

**Review by ishantiw — COMMENTED** [review]
LGTM, just a few comments

**Review by ishantiw — APPROVED** [review]
🚀

**Pull request merged**